src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.OperationSelector;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private static final String CANT_FIND_OPERATOR_MESSAGE = "Can't find such operator ";
+    private Map<Operation, OperationSelector> strategies;
+
+    @Override
+    public void provideStrategyList(Map<Operation, OperationSelector> strategies) {
+        this.strategies = strategies;
+    }
+
+    @Override
+    public OperationSelector get(String type) {
+        Operation requiredOperation = null;
+        for (Operation operation : Operation.values()) {
+            if (operation.getOperation().equals(type)) {
+                requiredOperation = operation;
+            }
+        }
+        if (requiredOperation == null) {
+            throw new RuntimeException(CANT_FIND_OPERATOR_MESSAGE + type);
+        }
```

Comment:
You can simplify this solution and move this logic to enum class. look at the method fromString on screenshot
<img width="675" alt="Screenshot 2022-02-18 at 10 48 23" src="https://user-images.githubusercontent.com/33026471/154649301-d60410df-6344-4e08-8d14-207694aaafb0.png">